clients/lsp-csharp.el

```diff
@@ -235,6 +241,99 @@ tarball or a zip file (based on a current platform) to TARGET-DIR."
   "Resolves path and arguments to use to start the server."
   (list (lsp-csharp--language-server-path) "-lsp"))
 
+(defun lsp-csharp--reset-test-buffer (present-buffer)
+  "Create new or reuse an existing test result output buffer.
+
+PRESENT-BUFFER will make the buffer be presented to the user."
+  (let ((existing-buffer (get-buffer lsp-csharp-test-run-buffer-name))
+        (project-root-dir (lsp--suggest-project-root)))
+    (if existing-buffer
+        (progn
+          (with-current-buffer existing-buffer
+            (setq buffer-read-only nil)
+            (erase-buffer)
+            (setq buffer-read-only t)
+            (setq default-directory project-root-dir))
+          existing-buffer)
+      (let ((buffer (get-buffer-create lsp-csharp-test-run-buffer-name)))
+        (with-current-buffer buffer
+          (setq default-directory project-root-dir)
```

Comment:
This has no effect, because `compilation-mode` is a minor-mode and as such deletes all local bindings within the buffer upon being enabled.
```emacs-lisp
(with-temp-buffer (setq-local default-directory t) (compilation-mode) default-directory) ; = whatever the default value of `default-directory` is. `setq-local` is for illustration here only, since that variable is defined buffer-local
```